### PR TITLE
Fix scanning of events for event triggers.

### PR DESCRIPTION
**Pull request by nagarajdivine**
## Description

* Fix scanning of events for event triggers.
closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting)).
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added relevant changelog entries under `.changelog/unreleased` (see [Adding Changes](https://github.com/provenance-io/provenance/blob/main/.changelog/README.md#adding-changes)).
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed event trigger detection so matching events properly queue triggers and duplicates are suppressed, preventing repeated executions.

* **Tests**
  * Added comprehensive tests verifying transaction event attribute matching, suppression behavior, and edge cases.

* **Changelog**
  * Updated release notes to document the event trigger processing fix.

[Review Change Stack](https://app.coderabbit.ai/change-stack/provenance-io/provenance/pull/2733?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: .changelog/unreleased/bug-fixes/trigger-tx-event-detection-suppression.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+* Fix scanning of events for event triggers.
+* Ensure correct queuing of matching events for trigger execution.
```
Comment (SpicyLemon) [inline .changelog/unreleased/bug-fixes/trigger-tx-event-detection-suppression.md]: These need links to the PR.

File: .changelog/unreleased/bug-fixes/trigger-tx-event-detection-suppression.md [hunk 2]
```diff
@@ -0,0 +1 @@
+* Fix event trigger processing by correcting event scanning and ensuring proper queuing of matching events for trigger execution.[PR 2733](https://github.com/provenance-io/provenance/pull/2733)
```
[the remark this answers is not in the document]
Comment (SpicyLemon) [reply .changelog/unreleased/bug-fixes/trigger-tx-event-detection-suppression.md]: To add to this, please also move the period to after the PR link.

**Review by SpicyLemon — APPROVED** [review]
